index.js

```diff
@@ -63,19 +63,25 @@ module.exports = postcss.plugin( 'postcss-flexboxfixer', function( opts ) {
         }
 
         var toColor;
-        for(var j = type === 'linear' ? 2 : 4; j < args.length; j++){
+        var startStep = type === 'linear' ? 2 : 4;
+        for (var j = startStep; j < args.length; j++) {
             var position, color, colorIndex;
             if(args[j].name === 'color-stop'){
                 position = args[j].args[0].name;
-                colorIndex = 1;
+                if (args[j].args[1]) {
+                    colorIndex = 1;
+                } else {
+                    colorIndex = 0;
+                    position = (j - startStep) / (args.length - startStep - 1);
```

Comment:
in this case:
```CSS
.direction {
    background: -webkit-gradient(linear, left top, right bottom, from(black), color-stop(rgba(0, 0, 0, 0.5)), to(white));
}
```
colorIndex should be `0`.
if not, this line will throw a Error
```JavaScript
color = args[j].args[colorIndex].name;
```